src/asset.rs

```diff
@@ -888,7 +884,7 @@ mod tests {
 
         // The extend method expects unique assets - adding duplicates would violate
         // the debug assertion, so this test verifies the normal case
-        asset_pool.extend(std::iter::empty::<AssetRef>());
+        asset_pool.extend(Vec::<AssetRef>::new());
```

Comment:
I think this will work:

```suggestion
        asset_pool.extend(Vec::new());
```